src/components/Carousel/Carousel.jsx

```diff
@@ -0,0 +1,239 @@
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex + 1,
+      }));
+      return;
+    }
+
+    this.setState(prevState => ({
+      left: prevState.left - parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex + 1,
+    }));
+  };
+
+  verifyLastItem = (lengthOfChildren, widthOfEachItem) => {
+    const { left } = this.state;
+    const finalLength = (lengthOfChildren - 1) * parseInt(widthOfEachItem, 10);
+
+    return left === -finalLength;
+  };
+
+  renderItems = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const items = children.map((child, index) => (
+      <MainListItem key={index} aria-hidden={currentSlideIndex !== index}>
+        {child}
+      </MainListItem>
+    ));
+
+    return items;
+  };
+
+  renderAnnouncement = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const content = (
+      <Announcement aria-live="polite" aria-atomic="true" tabIndex={-1}>
+        Slide
+        {currentSlideIndex + 1}
+        out of
+        {children.length}
+      </Announcement>
+    );
+
+    return content;
+  };
+
+  goToSlide = index => {
+    const { width } = this.props;
+    const newLeft = parseInt(width, 10) * index;
+    this.setState({
+      left: -newLeft,
+      currentSlideIndex: index,
+    });
+  };
+
+  render() {
+    const { left, currentSlideIndex } = this.state;
+    const { children, width } = this.props;
+    return (
+      <CarouselStyled>
+        {this.renderAnnouncement()}
+        <ButtonOverlapContainer left>
+          <Button
+            type="secondary"
+            icon={<ArrowLeft />}
+            hasIconOnly
+            onClick={() => {
+              this.moveBackward();
+            }}
+            label="Backwards"
+          />
+        </ButtonOverlapContainer>
+        <Content>
+          <Window width={width}>
+            <MainList left={left}>{this.renderItems()}</MainList>
+          </Window>
+          <IndicatorList>
+            {children.map((child, index) => (
+              <IndicatorListItem key={index}>
+                <IndicatorButton
+                  type="button"
+                  onClick={() => this.goToSlide(index)}
+                  active={index === currentSlideIndex}
+                />
+              </IndicatorListItem>
+            ))}
+          </IndicatorList>
+        </Content>
+        <ButtonOverlapContainer>
+          <Button
+            type="secondary"
+            icon={<ArrowRight />}
+            hasIconOnly
+            onClick={() => {
+              this.moveForward();
+            }}
+            label="Forwards"
+          />
+        </ButtonOverlapContainer>
+      </CarouselStyled>
+    );
+  }
+}
+
+Carousel.defaultProps = {
+  children: [],
+  width: '200px',
+};
+
+Carousel.propTypes = {
+  /** The content within the carousel */
+  children: PropTypes.array.isRequired,
```

Comment:
Just wanna double check that this is the right propType 😅

Reply:
from your usage I think that this should just be the standard `PropTypes.node` as it could be anything in theory, it could be a single node, or multiple nodes, or a string, or anything technically. 

If you want to limit what can go into here, you have some options, you could create the concept of a CarouselCard and export that, then you could set children to `children: PropTypes.oneOfType(CarouselCard)` but I would suggest against that as it can cause other issues when importing this library later from external projects, its probably easier to just allow any node element here for simplicity, but perhaps suggest limitations about its usage instead.

Reply:
Totally agree with @karelm let's keep this `PropTypes.node` for sake of simplicity and flexibility.

Reply:
gotcha, thanks <3